simple_certmanager/models.py

```diff
@@ -16,6 +22,107 @@
 from .validators import PrivateKeyValidator, PublicCertValidator
 
 
+class SigningRequest(DeleteFileFieldFilesMixin, models.Model):
+    country_name = models.CharField(
+        max_length=2, help_text=_("Two-letter country code")
+    )
+    organization_name = models.CharField(
+        max_length=100, help_text=_("The name of the organization")
+    )
+    state_or_province_name = models.CharField(
+        max_length=100, help_text=_("The state or province name")
+    )
+    email_address = models.EmailField(help_text=_("Email address for the certificate"))
+    common_name = models.CharField(
+        max_length=100, help_text=_("The common name for the certificate")
+    )
+    private_key = PrivateMediaFileField(
+        _("private key"),
+        help_text=_("The content of the private key"),
+        blank=True,
+        upload_to="ssl_certs_keys/%Y/%m/%d",
+        validators=[PrivateKeyValidator()],
+    )
+    csr = models.TextField(editable=False, blank=True)
+
+    _private_key_obj: PrivateKeyTypes | None = None
+
+    def save(self, *args, **kwargs):
+        if not self.private_key:
+            # Generate private key
+            private_key = rsa.generate_private_key(
+                public_exponent=65537, key_size=2048, backend=default_backend()
+            )
+            private_key_file_bytes = private_key.private_bytes(
+                encoding=serialization.Encoding.PEM,
+                format=serialization.PrivateFormat.PKCS8,
+                encryption_algorithm=serialization.NoEncryption(),
+            )
+            private_key_file = ContentFile(
+                private_key_file_bytes, name="private_key.pem"
+            )
+            self.private_key = private_key_file
+
+        if not self.csr:
+            # Generate CSR
+            csr_builder = x509.CertificateSigningRequestBuilder()
+            csr_builder = csr_builder.subject_name(
+                x509.Name(
+                    [
+                        x509.NameAttribute(NameOID.COUNTRY_NAME, self.country_name),
+                        x509.NameAttribute(
+                            NameOID.STATE_OR_PROVINCE_NAME, self.state_or_province_name
+                        ),
+                        x509.NameAttribute(
+                            NameOID.LOCALITY_NAME, self.organization_name
+                        ),
+                        x509.NameAttribute(
+                            NameOID.ORGANIZATION_NAME, self.organization_name
+                        ),
+                        x509.NameAttribute(NameOID.COMMON_NAME, self.common_name),
+                    ]
+                )
+            )
+
+            if self.email_address:
+                csr_builder = csr_builder.add_extension(
+                    x509.SubjectAlternativeName([x509.RFC822Name(self.email_address)]),
+                    critical=False,
+                )
```

Comment:
Depends on what the subject is.. if the certificate identifies a person it can be an email address.
It can be used as an alternative to PGP for email for instance... But in our usecase it's probably a service or organisation. For the former, URI or hostnames make more sense.